general-concepts/ebuild-revisions/text.xml

```diff
@@ -85,7 +85,11 @@ of thumb could be used as a guideline:
   <li>
     adding a patch to fix a build-time issue that prevented users from
     building the package (since it does not affect users who already
-    managed to build it),
+    managed to build it) unless: it affected runtime behaviour in some way
+    (e.g. <c>-Wformat</c> or <c>-Wimplicit-function-declaration</c> fixes);
```

Comment:
It's not purely cosmetic (despite it often seeming so, although some cases are):
- e.g. If a function returns a _Bool on amd64 but we assume it's an int, half of its bits get cleared.
- e.g. On arm64 (for Apple macs at least, with their different calling convention/ABI), there's been runtime corruption with passing arguments.

There's also been e.g. https://developers.redhat.com/blog/2019/04/22/implicit-function-declarations-flexs-use-of-reallocarray.

Florian covers it in his short talk at https://youtu.be/q5itHU2T5xU?t=2583 (but there's also a great LWN article: https://lwn.net/SubscriberLink/913505/b61bbc0c4296117f/).

All this to say: I wanted to call it out specifically because it's one of those things where something can _look_ like a pure build time fix, but actually affects code generation.